karabo_data/tests/test_agipd_geometry.py

```diff
@@ -22,23 +24,25 @@ def test_write_read_crystfel_file(tmpdir):
         quad_pos=[(-525, 625), (-550, -10), (520, -160), (542.5, 475)]
     )
     path = str(tmpdir / 'test.geom')
-    geom.write_crystfel_geom(path)
-
-    # We need to add some experiment details before cfelpyutils will read the
-    # file
-    with open(path, 'r') as f:
-        contents = f.read()
-    with open(path, 'w') as f:
-        f.write('clen = 0.119\n')
-        f.write('adu_per_eV = 0.0075\n')
-        f.write(contents)
+    geom.write_crystfel_geom(filename=path, photon_energy=9000,
+                             adu_per_ev=0.0075, clen=0.2)
```

Comment:
It would be good to have a test with each of the 3 detector classes, even if it's pretty basic. And at least one of them should be missing the photon_energy, adu_per_ev, clen parameters, so it exercises the code paths for when they have no value.

Reply:
Do we have a geometry test for dssc?

Reply:
Ah, of course. We do in master, but this branch split before it was added. I'll merge this and add a test separately.

Reply:
In fact, I don't think it even really makes sense for DSSC, because for now CrystFEL assumes square pixels. Even if we ignore the hexagonal pixels, you still have to treat them as rectangles to get them in roughly the right position.